pkg/assetvalidator/validateAssets.go

```diff
@@ -0,0 +1,69 @@
+package assetvalidator
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"time"
+
+	"github.com/asecurityteam/nexpose-asset-producer/pkg/domain"
+)
+
+// NexposeAssetValidator is used to validate a list of retrieved Assets
+type NexposeAssetValidator struct {
+}
+
+// ValidateAssets takes in a list of assets from Nexpose, and returns a list of valid Asset events, and a list of errors based
+// on defined validation rules
+func (v *NexposeAssetValidator) ValidateAssets(ctx context.Context, assets []domain.Asset, scanID string) ([]domain.AssetEvent, []error) {
+	assetEventListList := []domain.AssetEvent{}
+	errorList := []error{}
+	for _, asset := range assets {
+		scanTime, err := v.getScanTime(asset, scanID)
+		if err != nil {
+			errorList = append(errorList, err)
+			continue
+		}
+		assetEvent, err := v.assetPayloadToAssetEvent(asset, scanTime)
+		if err != nil {
+			errorList = append(errorList, err)
+			continue
+		}
+		assetEventListList = append(assetEventListList, assetEvent)
+	}
+	return assetEventListList, errorList
+}
+
+// GetScanTime searches through the asset's event history for a scan event with the ScanID
```

Comment:
nit: the initial capital G in this comment is wrong. I'm surprised this didn't get caught by your editor's static analysis.

Reply:
i think it's because un-exported functions don't need comments describing the function. Fixed